plugins/topology/src/utils/pod-resource-utils.test.ts

```diff
@@ -21,4 +47,91 @@ describe('PodResourceUtils', () => {
     expect(getDeploymentRevision(deployment)).toBe(null);
     expect(getDeploymentRevision({})).toBe(null);
   });
+
+  it('should return pods for a given Deployment', () => {
+    let podRCData = getPodsForDeployment(
+      mockKubernetesResponse.deployments[0] as V1Deployment,
+      mockResources,
+    );
+    expect(podRCData.pods).toHaveLength(3);
+    expect(podRCData.current).not.toBeNull();
+    expect(podRCData.previous).toBeFalsy();
+    expect(podRCData.isRollingOut).toBeFalsy();
+
+    podRCData = getPodsForDeployment({}, mockResources);
+    expect(podRCData.pods).toHaveLength(0);
+
+    podRCData = getPodsForDeployment(
+      mockKubernetesResponse.deployments[1] as V1Deployment,
+      mockResources,
+    );
+    expect(podRCData.pods).toHaveLength(1);
+
+    podRCData = getPodsForDeployment(
+      mockKubernetesResponse.deployments[2] as V1Deployment,
+      mockResources,
+    );
+    expect(podRCData.pods).toHaveLength(1);
+  });
+
+  it('should return pods for a given StatefulSet', () => {
+    let podRCData = getPodsForStatefulSet(
+      mockKubernetesResponse.statefulsets[0] as V1StatefulSet,
+      mockResources,
+    );
+    expect(podRCData.pods).toHaveLength(1);
+    expect(podRCData.current).not.toBeNull();
+    expect(podRCData.previous).toBeFalsy();
+    expect(podRCData.isRollingOut).toBeFalsy();
+
+    podRCData = getPodsForStatefulSet({}, mockResources);
+    expect(podRCData.pods).toHaveLength(0);
+
+    mockKubernetesResponse.statefulsets = [];
+    podRCData = getPodsForStatefulSet(
+      mockKubernetesResponse.statefulsets[0] as V1StatefulSet,
+      mockResources,
+    );
+    expect(podRCData.pods).toHaveLength(0);
```

Comment:
title says `should return pods for a given StatefulSet` but assertion checks otherwise?

Reply:
updated the title to "should return pods when StatefulSet exists" as here I am testing multiple StatefulSet scenarios